sunpy/map/mapbase.py

```diff
@@ -1687,20 +1686,13 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=3, scale=1.0,
         unpad_y = -np.min((diff[1], 0))
 
         # Pad the image array
-
-        if issubclass(self.data.dtype.type, numbers.Integral) and (missing % 1 != 0):
-            warn_user("The specified `missing` value is not an integer, but the data "
-                      "array is of integer type, so the output may be strange.")
```

Comment:
So with numpy >= 1.2.0, this warning will be raised and then the pad below will error.

So I think we can remove this now or maybe add a check for older versions of numpy ?